Update Disable-InactiveADAccounts.ps1

Average-Bear opened the pull request:
NOTE: At the current time, I have NOT TESTED the proposed changes yet. I've simply restructured a few pieces under the view of "this should work better"; e.g. [PSCustomObject] vs. Building the object with Add-Member.

PLEASE, test before taking on these changes; there may be errors. Let's collaborate on how to make this thing as fast as it should be.

Commit:
Added delay to DFSR auto recovery to improve success rate.

Average-Bear committed:
Update Disable-InactiveADAccounts.ps1

NOTE: I the current time, I have NOT TESTED the proposed changes yet. I've simply restructed a few pieces under the view of "this would work better"; e.g. [PSCustomObject] vs. Building the object with Add-Member.

PLEASE, test before taking on these changes; there may be errors. Let's collaborate on how to make this thing as fast as it should be.

Average-Bear committed:
Update Disable-InactiveADAccounts.ps1

Average-Bear reviewed (commented):
Commented out the Write-Output while testing Write-Progress portion.

AndrewEllis93 commented:
Thanks man. I incorporated some of your proposed changes, namely the PSCustomObject which led me down the road of moving everything over to hash tables... I found out getting the AD objects from the arrays was where it was hanging up, even when getting them directly using the index number. Basically I followed that same vein of thought and converted the AD users results to hash tables and the subsequent processing is a LOT faster. The only downside is that I have to convert everything to hash tables first, which takes a while on its own. 

Your edited copy didn't quite work as-is so I picked some bits from it to use in my revamped version. I will probably go back and grab some more snippets from yours as I find more time to work on it. Cheers.

Average-Bear commented:
There were tons of things I picked up from the PowerShell Conference 2017
videos. Those guys are incredible.

One of the biggest things was that all variables are natively an array,
which is why you can shove $var in front of your foreach loop and it
becomes all results. I have several scripts and functions.laying around
that are codes just like I see some of yours and after some tinkering some
time later I learned against it.

The better part of this is that we take steps towards a process, even if
the script takes an hour to run, It fucking works and that is the real
prize. Time saving comes later.

You're welcome to get any ideas from my work on GitHub and ask any
questions/collaborate. I'm not the gold standard of coding by any stretch
of the imagination but, I like developing solutions.

On Dec 31, 2017 14:30, "AndrewEllis93" <notifications@github.com> wrote:

> Thanks man. I incorporated some of your proposed changes, namely the
> PSCustomObject which led me down the road of moving everything over to hash
> tables... I found out getting the AD objects from the arrays was where it
> was hanging up, even when getting them directly using the index number.
> Basically I followed that same vein of thought and converted the AD users
> results to hash tables and the subsequent processing is a LOT faster. The
> only downside is that I have to convert everything to hash tables first,
> which takes a while on its own.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/AndrewEllis93/PowerShell-Scripts/pull/1#issuecomment-354581307>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AXj5SUIIaWxFEDndLS47dCnEaFS638Eyks5tFvHggaJpZM4RPd24>
> .
>
